tests/filters/ignore_new_executors_test.cpp

```diff
@@ -0,0 +1,123 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "mesos/mesos.hpp"
+
+#include "filters/ignore_new_executors.hpp"
+
+#include "tests/common/serenity.hpp"
+#include "tests/common/sinks/dummy_sink.hpp"
+#include "tests/common/sources/json_source.hpp"
+
+#include "tests/common/sinks/printer_sink.hpp"
+
+namespace mesos {
+namespace serenity {
+namespace tests {
+
+using testing::Return;
+using testing::Sequence;
+
+constexpr int IGNORE_NEW_EXECUTORS_SAMPLES = 6;
+
+class MockIgnoreNewExecutorsFilter : public IgnoreNewExecutorsFilter {
+ public:
+  explicit MockIgnoreNewExecutorsFilter(uint32_t arg) :
+      IgnoreNewExecutorsFilter(arg) {}
+
+  MOCK_METHOD1(GetTime, time_t(time_t* arg));
+};
+
+/**
+ * Time is mocked to be one second after begining of an Epoch
+ */
+TEST(IgnoreNewExecutorsFilter, IgnoreAllExecutors) {
+  DummySink<ResourceUsage> dummySink;
+  PrinterSink<ResourceUsage> printer;
+  MockIgnoreNewExecutorsFilter filter(200);
+  JsonSource jsonSource;
+
+  jsonSource.addConsumer(&filter);
+  filter.addConsumer(&dummySink);
+
+  EXPECT_CALL(filter, GetTime(nullptr)).Times(4).WillRepeatedly(Return(1));
+
+  jsonSource.RunTests("tests/fixtures/"
+                      "baseline_smoke_test_resource_usage.json");
+
+  ASSERT_EQ(dummySink.numberOfMessagesConsumed, 0);
+}
+
+/**
+ * Time is mocked to be one second before end of an Epoch
+ */
+TEST(IgnoreNewExecutorsFilter, PassAllExecutors) {
+  DummySink<ResourceUsage> dummySink;
+  MockIgnoreNewExecutorsFilter filter(0);
+  JsonSource jsonSource;
+
+  jsonSource.addConsumer(&filter);
+  filter.addConsumer(&dummySink);
+
+  EXPECT_CALL(filter, GetTime(nullptr)).Times(4).
+      WillRepeatedly(Return(2147483647));
```

Comment:
If we want to be consistent with Mesos we should stick dot to the next item. E.g:

```
EXPECT_CALL(filter, GetTime(nullptr))
    .Times(4)
    .WillRepeatedly(Return(2147483647));
```